encoder_map_test.go

```diff
@@ -63,6 +63,20 @@ func TestEncoder_MapOfStruct(t *testing.T) {
 	assert.Equal(t, []byte{0x01, 0x12, 0x06, 0x66, 0x6F, 0x6F, 0x36, 0x06, 0x66, 0x6f, 0x6f, 0x0}, buf.Bytes())
 }
 
+func TestEncoder_MapOfUnionBool(t *testing.T) {
```

Comment:
This test is in the wrong place. It is confusing to test a union of `interface{}` in the map tests. Could you move them to the `*_union_test.go` file and follow the tests like `TestDecoder_UnionInterface`?